### PR TITLE
Add code to compare embedding quality on role prediction problem

### PR DESCRIPTION
**Pull request by zurk**

### CODE REVIEW

File: sourced/ml/__main__.py [hunk 1]
```diff
@@ -318,6 +318,20 @@ def add_parser(name, help_message):
         "-f", "--features", nargs="+",
         choices=[ex.NAME for ex in extractors.__extractors__.values()],
         default=None, help="To keep only specific features, if not specified all will be kept.")
+    # ------------------------------------------------------------------------
+    id2role_eval = add_parser("id2role-eval",
+                              "Compare embedding quality on role prediction problem.")
```
Comment (vmarkovtsev) [inline sourced/ml/__main__.py]: the embeddings

File: sourced/ml/cmd/id2role_eval.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    del df
+    rare = set(vc[vc < 10].index)
+    log.info("{} rare roles excluded. ".format(len(rare)))
+    df_unique = df_unique[~df_unique["role"].isin(rare)]
+    log.debug("Convert words to its embeddings")
+    Xs, y = identifiers_to_datasets(df_unique, models, log)
+
+    final_report.append("Pairs number: {}.\n".format(len(valid_tokens)))
+    final_report.append("| embedding name            | score            | best C value |")
+    final_report.append("|---------------------------|------------------|--------------|")
```
Comment (vmarkovtsev) [inline sourced/ml/cmd/id2role_eval.py]: :-(

1. use pandas.DataFrame to format the table OR
2. use https://github.com/foutaise/texttable

File: sourced/ml/cmd/id2role_eval.py [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+    # Count identifiers in dataset
+    log.info("Have embeddings only for {} tokens from {} in your dataset".format(
+        len(valid_tokens), len(df_ids)))
+    df_unique = df.groupby("identifier").agg(lambda x: x.value_counts().index[0])
+    df_unique["identifier"] = df_unique.index
+    # Exclude rare roles
+    vc = df["role"].value_counts()
+    del df
+    rare = set(vc[vc < 10].index)
+    log.info("{} rare roles excluded. ".format(len(rare)))
```
Comment (vmarkovtsev) [inline sourced/ml/cmd/id2role_eval.py]: we do not use `format()`: https://github.com/src-d/guide/blob/master/engineering/conventions-python.md#code-style

**Comment by zurk** [comment]
Are we ready to merge this one too, @vmarkovtsev ?

**Pull request merged**